treeherder/etl/classification_mirroring.py

```diff
@@ -27,25 +26,16 @@ def generate_request_body(self):
         """
         Create the data structure that will be sent to Elasticsearch.
         """
-        with JobsModel(self.project) as jobs_model, ArtifactsModel(self.project) as artifacts_model:
+        with JobsModel(self.project) as jobs_model:
             job_data = jobs_model.get_job(self.job_id)[0]
             buildtype = " ".join(sorted(OptionCollection.objects.values_list(
                 'option__name', flat=True).filter(
                     option_collection_hash=job_data["option_collection_hash"])))
             revision_list = jobs_model.get_resultset_revisions_list(job_data["result_set_id"])
-            buildapi_artifact = artifacts_model.get_job_artifact_list(0, 1, {
-                'job_id': set([("=", self.job_id)]),
-                'name': set([("=", "buildapi")])
-            })
-            if buildapi_artifact:
-                buildname = buildapi_artifact[0]["blob"]["buildername"]
-            else:
-                # OrangeFactor needs a buildname to be set or it skips the failure
-                # classification, so we make one up for non-buildbot jobs.
-                buildname = 'non-buildbot %s test %s' % (job_data["platform"], job_data["job_type_name"])
+            ref_data_name = job_data["ref_data_name"]
```

Comment:
job_detail doesn't have a buildername associated with it, we now rely on ref_data_name for this.

From what I see of the code you linked to, I don't think we use buildname at all for taskcluster -- we would just fall back to using the "type" field (which maps to something like "B2G Emulator Image Build"):

https://hg.mozilla.org/automation/orangefactor/file/tip/server/handlers.py#l270

So I think the current patch already has pretty much the optimal behaviour. OrangeFactor will use the information in the buildername if there is one now, falling back to other information in the job blob if it isn't intelligible.

Does that make sense?